Sources/Core/BasePresenter.swift

```diff
@@ -0,0 +1,131 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+import UIKit
+
+open class BasePresenter: OnboardingPresenter {
+
+    public typealias OnboardingData = (titleText: String, bodyText: String, image: UIImage?)
+
+    // MARK: Customizable properties
+    public var cellBackgroundColor: UIColor = .orange
+    public var doneButtonColor: UIColor = .white
+    public var doneButtonTextColor: UIColor = .black
+    public var textColor: UIColor = .white
+    public var swipeLabelText = ""
+    public var titleFont: UIFont = UIFont.boldSystemFont(ofSize: 24)
+    public var bodyFont: UIFont = UIFont.systemFont(ofSize: 18)
+    public var model: [OnboardingData] = [
+        (titleText: "Title 1",
+         bodyText: "Subtitle 1",
+         image: nil),
+        (titleText: "Title 2",
+         bodyText: "Subtitle 2",
+         image: nil),
+        (titleText: "Title 3",
+         bodyText: "Subtitle 3",
+         image: nil)
+        ]
+
+    // MARK: OnboardingPresenter protocol
+    public var onOnBoardingFinished: (() -> ())?
+    public var onOnboardingSkipped: (() -> ())?
+
+    public var cellProviders: [CellProvider] {
+        return [.nib(
+                    name:"OnboardingCell",
+                    identifier: OnboardingCell.reuseIdentifier,
+                    bundle: Resources.bundle
+                )]
+    }
+
+    public var pageCount: Int {
+        return model.count
+    }
+
+    public init() {}
+
+    open func visibilityChanged(for cell: UICollectionViewCell, at index: Int, amount: CGFloat) {
+        guard let cell = cell as? OnboardingCell, index == pageCount - 1  else { return }
+        cell.doneButtonBottomConstraint.constant = 60 * amount
+        cell.setNeedsLayout()
+        cell.layoutIfNeeded()
+    }
+
+    open func style(pageControl: UIPageControl?) {
+        pageControl?.backgroundColor = .clear
+        pageControl?.numberOfPages = pageCount
+        pageControl?.pageIndicatorTintColor = .lightGray
+        pageControl?.currentPageIndicatorTintColor = .white
+    }
+
+    open func style(collection: UICollectionView?) { }
+
+    open func style(skip: UIButton?) {
+        skip?.setTitle("Skip", for: .normal)
+        skip?.setTitleColor(.white, for: .normal)
+    }
+
+    open func reuseIdentifier(for page: Int) -> String {
+        switch cellProviders.first! {
```

Comment:
Yes, but in this case, since the BasePresenter just have 1 type of cell, I just return it always.